pkg/vsphere/extraconfig/encode.go

```diff
@@ -252,6 +253,9 @@ func encodeWithPrefix(src interface{}, prefix string) []types.BaseOptionValue {
 				switch field.Interface().(type) {
 				case time.Time:
 					config = append(config, &types.OptionValue{Key: key, Value: field.Interface().(time.Time).String()})
+				case net.IPNet:
+					n := field.Interface().(net.IPNet)
+					config = append(config, &types.OptionValue{Key: key, Value: n.String()})
```

Comment:
Not sure if it would make your life easier, but if you don't special case this for `net.IPNet` and instead reflect down to `byte` and cast it to a string, on the other side you won't need the special case which calls `ParseCIDR`.

In otherwords, don't use IPNet.String() and instead stringify the members directly (`IP` and `IPMask` are both of type `byte`)